File: LeetCode/Math/JS/Leetcode780-到达终点.js
var reachingPoints = function (sx, sy, tx, ty) {
    while (tx > sx && ty > sy) {
        if (tx > ty) tx %= ty;
        else ty %= tx;
    }
    if (tx < sx || ty < sy) return false;
    if (tx == sx) return (ty - sy) % sx == 0;
    return (tx - sx) % sy == 0;
};

File: LeetCode/Math/Golang/Leetcode780-到达终点.go
package sbgo

func reachingPoints(sx int, sy int, tx int, ty int) bool {
	for tx > sx && ty > sy {
		if tx > ty {
			tx %= ty
		} else {
			ty %= tx
		}
	}

	if tx < sx || ty < sy {
		return false
	}
	if sx == tx {
		return (ty-sy)%tx == 0
	}
	return (tx-sx)%sy == 0
}
